core/templates/dev/head/pages/collection_editor/CollectionEditorNavbarDirective.js

```diff
@@ -68,11 +78,15 @@ oppia.directive('collectionEditorNavbar', [function() {
           // action since it is not reversible.
           CollectionRightsBackendApiService.setCollectionPublic(
             $scope.collectionId, $scope.collection.getVersion()).then(
-            function() {
-              // TODO(bhenning): There should be a scope-level rights object
-              // used, instead. The rights object should be loaded with the
-              // collection.
-              $scope.isPrivate = false;
+            function(data) {
+              if (!data.is_private) {
+                $scope.isPrivate = data.is_private;
+                $rootScope.$broadcast(EVENT_COLLECTION_STATUS_CHANGE);
+              } else {
+                $log.error(
```

Comment:
Actually, maybe this check should go in setCollectionPublic()? Ditto below for setCollectionPrivate().